src/main/java/com/nkh/ECommerceShop/service/CartsService.java

```diff
@@ -0,0 +1,54 @@
+package com.nkh.ECommerceShop.service;
+
+import com.nkh.ECommerceShop.exception.NotEnoughProductQuantityException;
+import com.nkh.ECommerceShop.model.Cart;
+import com.nkh.ECommerceShop.model.CartProduct;
+import com.nkh.ECommerceShop.model.Product;
+import com.nkh.ECommerceShop.repository.CartsRepository;
+import com.nkh.ECommerceShop.security.service.UserDetailsImpl;
+import jakarta.transaction.Transactional;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Service;
+
+import java.util.Optional;
+
+@Service
+public class CartsService {
```

Comment:
adding unit test for services is highly recommended

Reply:
I have separate ticket for tests for each userstory